### PR TITLE
add admin limitation for update floorName and tag

### PR DESCRIPTION
**Pull request by Jo-Zh**
- added user admin role check to render edit form.
- to test:
'run_project_docker.sh'
'cd client && yarn cypress:run:local'  - should all passed except minimapTest, will need to fix in another ticket.

### CODE REVIEW

File: client/cypress/e2e/MinimapTest.cy.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ testEachZone((zone: Cypress.PrismZone) => {
     it(`Testing: user changes x coordinates input in the form, the targeted mininode position changes correctly`, () => {
       if (zone.adminUser) {
         cy.wait(getReqAlias).then(() => {
-          const randX = Math.floor(Math.random() * 9) * 10 + 10;
+          const randX = 50;
```
Comment (anzau) [inline client/cypress/e2e/MinimapTest.cy.ts:47]: Why is this changed when we're getting a random value for X?

File: client/cypress/e2e/MinimapTest.cy.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ testEachZone((zone: Cypress.PrismZone) => {
     it(`Testing: user changes y coordinates input in the form, the targeted mininode position changes correctly`, () => {
       if (zone.adminUser) {
         cy.wait(getReqAlias).then(() => {
-          const randY = Math.floor(Math.random() * 9) * 10 + 10;
+          const randY = 1;
```
Comment (anzau) [inline client/cypress/e2e/MinimapTest.cy.ts:126]: Same here...

File: client/src/components/FloorDetailsForm.tsx [hunk 3]
```diff
@@ -32,9 +32,9 @@ const FloorDetailsForm: React.FC<FloorDetailsFormProps> = ({
   floorTagState,
   submitVisibilityState,
 }): JSX.Element => {
-  if (showForm === undefined) showForm = true;
+  const [user] = useUserContext();
+  if (showForm === undefined && user?.isAdmin) showForm = true;
```
Comment (anzau) [inline client/src/components/FloorDetailsForm.tsx:36]: Maybe we can shorten this condition by:

`if (!showForm && user?.isAdmin)`

Another way could be:

`showForm = !showForm && user?.isAdmin` (Assuming they are meant to equal to true)

Comment (Jo-Zh) [reply client/src/components/FloorDetailsForm.tsx]: fixed...

File: client/cypress/e2e/MinimapTest.cy.ts (hunk 2 above)
Comment (Jo-Zh) [reply client/cypress/e2e/MinimapTest.cy.ts]: fixed, passed locally, but with many trials

File: client/cypress/e2e/MinimapTest.cy.ts (hunk 1 above)
Comment (Jo-Zh) [reply client/cypress/e2e/MinimapTest.cy.ts]: fixed, passed locally, but with many trials

File: client/src/components/FloorDetailsForm.tsx (hunk 3 above)
Comment (freshc0w) [reply client/src/components/FloorDetailsForm.tsx]: Hi, I know I'm not in work but just saw this and felt that I needed to comment real quick
`if (!showForm && user?.isAdmin)` will not work as intended since showForm is meant to be a boolean. There are some cases where the `showForm` prop can be false and in this case, we **don't want** the form to be shown. The reason the condition is `=== undefined` is that when the prop (which is an _optional_ prop when not given as the component argument, it then default set it to `true`.

Highly suggest the component to be `showForm === undefined`

Comment (Jo-Zh) [reply client/src/components/FloorDetailsForm.tsx]: seems, even if showForm true, When user is not admin, the form will not display.... is what expected...

> Hi, I know I'm not in work but just saw this and felt that I needed to comment real quick `if (!showForm && user?.isAdmin)` will not work as intended since showForm is meant to be a boolean. There are some cases where the `showForm` prop can be false and in this case, we **don't want** the form to be shown. The reason the condition is `=== undefined` is that when the prop (which is an _optional_ prop i.e. when not given as the component argument) is not provided, it then will default set it to `true`, showing the form regardless.
> 
> Highly suggest the component to be reverted back to `showForm === undefined`

File: client/src/components/FloorDetailsForm.tsx [hunk 4]
```diff
@@ -32,9 +32,9 @@ const FloorDetailsForm: React.FC<FloorDetailsFormProps> = ({
   floorTagState,
   submitVisibilityState,
 }): JSX.Element => {
-  if (showForm === undefined) showForm = true;
+  const [user] = useUserContext();
+  if (!showForm && user?.isAdmin) showForm = true;
```
Comment (anzau) [inline client/src/components/FloorDetailsForm.tsx:36]: Maybe I would try to simplify these 2 lines because the condition is being repeated:

```typescript
if (!showform){
   showForm = user?.isAdmin
   return <></>;
}
```

Comment (Jo-Zh) [reply client/src/components/FloorDetailsForm.tsx]: fixed

**Review by anzau — APPROVED** [review]
